source/server/connection_handler_impl.cc

```diff
@@ -107,6 +112,66 @@ void ConnectionHandlerImpl::enableListeners() {
   }
 }
 
+void ConnectionHandlerImpl::closeSocketsOnListenerUpdate(
+    const Network::ListenerConfig& old_config, const Network::ListenerConfig& new_config) {
+  const std::string& listener_name = new_config.name();
+  ENVOY_LOG(debug,
+            "close sockets that the old listener {} has stored if the updated listener does not "
+            "contain the same filter chain.",
+            listener_name);
+  for (auto& [filter_chain_message, listener_sockets_map] : pending_sockets_) {
+    for (auto& [name, socket_metadata_pairs] : listener_sockets_map) {
+      if (name == listener_name) {
```

Comment:
Why do you iterate over the entire map to find an exact key?

Reply:
Updated the key of ```pending_sockets_``` from two levels ```[filter_chain][listener_name]``` to one level ```[pair(filter_chain, listener_name)]```